src/librustc/hir/mod.rs

```diff
@@ -539,9 +538,10 @@ pub enum PatKind {
     /// The `bool` is `true` in the presence of a `..`.
     Struct(Path, HirVec<Spanned<FieldPat>>, bool),
 
-    /// A tuple struct/variant pattern `Variant(x, y, z)`.
-    /// "None" means a `Variant(..)` pattern where we don't bind the fields to names.
-    TupleStruct(Path, Option<HirVec<P<Pat>>>),
+    /// A tuple struct/variant pattern `Variant(x, y, .., z)`.
+    /// If the `..` pattern fragment presents, then `Option<usize>` denotes its position.
```

Comment:
Not that there is anything wrong with it, but I am surprised by this choice; I would have expected `Vec<P<Pat>>` and `Option<Vec<P<Pat>>>`, where the second set of patterns come after the `..`. Did you consider this and find that the current optional usize works out better (or just that you prefer it)? Anyway, it seems ok as is, just wanted to toss an alternative out there.